src/components/Counter.js

```diff
@@ -1,63 +1,51 @@
 import React, { Component } from "react";
-import { connect } from 'react-redux';
-import { increment, decrement } from '../actions';
+import { connect } from "react-redux";
+import { increment, decrement } from "../actions";
 
 class Counter extends Component {
-    incrementIfOdd = () => {
-        // Extra: Implement an increment function that
-        // only increments if the counter value is odd
-    };
+  incrementIfOdd = () => {
+    // Extra: Implement an increment function that
+    // only increments if the counter value is odd
+    if (this.props.count % 2 === 1) this.props.increment();
+  };
 
-    incrementAsync = () => {
-        // Extra: Implement an increment function that
-        // increments after waiting for one second
-    };
+  incrementAsync = () => {
+    // Extra: Implement an increment function that
+    // increments after waiting for one second
+    setTimeout(this.props.increment, 1000);
+  };
 
-    render() {
-        // Fill in the two button onClick methods
-        // Upon clicking these buttons, the count
-        // should decrement or increment accordingly
-        return (
-            <p>
-                Clicked: {this.props.count} times
-                {" "}
-                <button onClick={() => /* Fill me in */ }>
-                    +
-                </button>
-                {" "}
-                <button onClick={() => /* Fill me in */ }>
-                    -
-                </button>
-                {" "}
-                 {/* Uncomment these button tags if you got
-                around to implementing the extra credit functions */}
-                {/* <button onClick={this.incrementIfOdd}>
-                    Increment if odd
-                </button>
-                {" "}
-                <button onClick={this.incrementAsync}>
-                    Increment async
-                </button>  */}
-            </p>
-        );
-    }
+  render() {
+    // Fill in the two button onClick methods
+    // Upon clicking these buttons, the count
+    // should decrement or increment accordingly
+    return (
+      <p>
+        Clicked: {this.props.count} times{" "}
+        <button onClick={() => this.props.increment()}>+</button>{" "}
```

Comment:
No need for callback here if immediately return function.  The function will be invoked automatically when the event is triggered.
`onClick={this.props.increment}`